File: js/src/02integer.js
import {
    nextInt
    , compareInt
    , initScanner
    , initChecker
    , inputsEOF
    , outputsEOF
} from './io.js'

(async () => {
    for (let taskNo = 1; taskNo <= 30; taskNo++) {
        const taskNoStr = String(taskNo).padStart(3, 0)
        for (let testNo = 1; testNo <= 100; testNo++) {
            const testNoStr = String(testNo).padStart(3, 0)
            const filePath = `../data/kt-gov2/02integer/Integer${taskNoStr}/${testNoStr}/${testNoStr}`

            await initScanner(filePath + '.dat')
            await initChecker(filePath + '.ans')

            let ok = true
            switch (taskNo) {
                case 1: ok = integer1(); break
                case 2: ok = integer2(); break
                case 3: ok = integer3(); break
                case 4: ok = integer4(); break
                case 5: ok = integer5(); break
                case 6: ok = integer6(); break
                case 7: ok = integer7(); break
                case 8: ok = integer8(); break
                case 9: ok = integer9(); break
                case 10: ok = integer10(); break
                case 11: ok = integer11(); break
                case 12: ok = integer12(); break
                case 13: ok = integer13(); break
                case 14: ok = integer14(); break
                case 15: ok = integer15(); break
                case 16: ok = integer16(); break
                case 17: ok = integer17(); break
                case 18: ok = integer18(); break
                case 19: ok = integer19(); break
                case 20: ok = integer20(); break
                case 21: ok = integer21(); break
                case 22: ok = integer22(); break
                case 23: ok = integer23(); break
                case 24: ok = integer24(); break
                case 25: ok = integer25(); break
                case 26: ok = integer26(); break
                case 27: ok = integer27(); break
                case 28: ok = integer28(); break
                case 29: ok = integer29(); break
                case 30: ok = integer30(); break
            }

            if (!ok) {
                console.log("Integer" + taskNoStr + " #" + testNoStr + " has NOT been tested!")
                return
            }

            if (!inputsEOF()) {
                console.log("Integer" + taskNoStr + " #" + testNoStr + " input NOT EOF!")
                return
            }

            if (!outputsEOF()) {
                console.log("Integer" + taskNoStr + " #" + testNoStr + " output NOT EOF!")
                return
            }
        }
        console.log("Integer" + taskNoStr + " has been tested!")
    }
    console.log("Integer has been tested!")
})()

const integer1 = () => {
    let l = nextInt()
    let meters = parseInt(l / 100)
    return compareInt(meters)
}

const integer2 = () => {
    let m = nextInt()
    let tons = parseInt(m / 1000)
    return compareInt(tons)
}

const integer3 = () => {
    let bytes = nextInt()
    let kBytes = parseInt(bytes / 1024)
    return compareInt(kBytes)
}

const integer4 = () => {
    let a = nextInt()
    let b = nextInt()
    let segmentsCount = parseInt(a / b)
    return compareInt(segmentsCount)
}

const integer5 = () => {
    let a = nextInt()
    let b = nextInt()
    let unusedSegment = a % b
    return compareInt(unusedSegment)
}

const integer6 = () => {
    let number = nextInt()
    let tens = parseInt(number / 10)
    let ones = number % 10
    return compareInt(tens, ones)
}

const integer7 = () => {
    let number = nextInt()
    let tens = parseInt(number / 10)
    let ones = number % 10
    let sum = tens + ones
    let pro = tens * ones
    return compareInt(sum, pro)
}

const integer8 = () => {
    let number = nextInt()
    let tens = parseInt(number / 10)
    let ones = number % 10
    let rebmun = ones * 10 + tens
    return compareInt(rebmun)
}

const integer9 = () => {
    let number = nextInt()
    let hundreds = parseInt(number / 100)
    return compareInt(hundreds)
}

const integer10 = () => {
    let number = nextInt()
    let ones = number % 10
    let tens = parseInt(number / 10) % 10
    return compareInt(ones, tens)
}

const integer11 = () => {
    let number = nextInt()
    let hundreds = parseInt(number / 100)
    let tens = parseInt(number / 10) % 10
    let ones = number % 10
    let sum = hundreds + tens + ones
    let pro = hundreds * tens * ones
    return compareInt(sum, pro)
}

const integer12 = () => {
    let number = nextInt()
    let rebmun = (number % 10) * 100 + (parseInt(number / 10) % 10) * 10 + parseInt(number / 100)
    return compareInt(rebmun)
}

const integer13 = () => {
    let number = nextInt()
    number = (number % 100) * 10 + parseInt(number / 100)
    return compareInt(number)
}

const integer14 = () => {
    let number = nextInt()
    number = number % 10 * 100 + parseInt(number / 10)
    return compareInt(number)
}

const integer15 = () => {
    let number = nextInt()
    number = parseInt(number / 10) % 10 * 100 + parseInt(number / 100) * 10 + number % 10
    return compareInt(number)
}

const integer16 = () => {
    let number = nextInt()
    number = number - number % 100 + number % 10 * 10 + parseInt(number / 10) % 10
    return compareInt(number)
}

const integer17 = () => {
    let number = nextInt()
    let hundreds = parseInt(number / 100) % 10
    return compareInt(hundreds)
}

const integer18 = () => {
    let number = nextInt()
    let thousands = parseInt(number / 1000) % 10
    return compareInt(thousands)
}

const integer19 = () => {
    let n = nextInt()
    let minutes = parseInt(n / 60)
    return compareInt(minutes)
}

const integer20 = () => {
    let n = nextInt()
    let hours = parseInt(n / 3600)
    return compareInt(hours)
}

const integer21 = () => {
    let n = nextInt()
    let seconds = n % 60
    return compareInt(seconds)
}

const integer22 = () => {
    let n = nextInt()
    let seconds = n % 3600
    return compareInt(seconds)
}

const integer23 = () => {
    let n = nextInt()
    let minutes = parseInt(n % 3600 / 60)
    return compareInt(minutes)
}

const integer24 = () => {
    let k = nextInt()
    let weekNo = k % 7
    return compareInt(weekNo)
}

const integer25 = () => {
    let k = nextInt()
    let weekNo = (k + 3) % 7
    return compareInt(weekNo)
}

const integer26 = () => {
    let k = nextInt()
    let weekNo = k % 7 + 1
    return compareInt(weekNo)
}

const integer27 = () => {
    let k = nextInt()
    let weekNo = (k + 4) % 7 + 1
    return compareInt(weekNo)
}

const integer28 = () => {
    let k = nextInt()
    let n = nextInt()
    let weekNo = (k + n - 2) % 7 + 1
    return compareInt(weekNo)
}

const integer29 = () => {
    let a = nextInt()
    let b = nextInt()
    let c = nextInt()
    let cInA = parseInt(a / c)
    let cInB = parseInt(b / c)
    let squaresCount = cInA * cInB
    let unusedPart = a * b - squaresCount * c * c
    return compareInt(squaresCount, unusedPart)
}

const integer30 = () => {
    let year = nextInt()
    let century = parseInt((year - 1) / 100) + 1
    return compareInt(century)
}

File: java/02integer.java
import java.util.Scanner;

class IntegerTask {

	private boolean integer1() {
		Scanner scanner = new Scanner(System.in);
		System.out.print("L = ");
		int L = scanner.nextInt();
		int meters = L / 100;
		System.out.println("meters = " + meters);
		return false;
	}

	private boolean integer2() {
		Scanner scanner = new Scanner(System.in);
		System.out.print("M = ");
		int M = scanner.nextInt();
		int tons = M / 1000;
		System.out.println("tons = " + tons);
		return false;
	}

	private boolean integer3() {
		Scanner scanner = new Scanner(System.in);
		System.out.print("bytes = ");
		int bytes = scanner.nextInt();
		int kBytes = bytes / 1024;
		System.out.println("kBytes = " + kBytes);
		return false;
	}

	private boolean integer4() {
		Scanner scanner = new Scanner(System.in);
		System.out.print("A = ");
		int A = scanner.nextInt();
		System.out.print("B = ");
		int B = scanner.nextInt();
		int BinA = A / B;
		System.out.println("B in A:\t" + BinA);
		return false;
	}

	private boolean integer5() {
		Scanner scanner = new Scanner(System.in);
		System.out.print("A = ");
		int A = scanner.nextInt();
		System.out.print("B = ");
		int B = scanner.nextInt();
		int freeSpace = A % B;
		System.out.println("free space:\t" + freeSpace);
		return false;
	}

	private boolean integer6() {
		Scanner scanner = new Scanner(System.in);
		System.out.print("number [10-99]:\t");
		int number = scanner.nextInt();
		int dahi = number / 10;
		int vohid = number % 10;
		System.out.println("dahi = " + dahi);
		System.out.println("vohid = " + vohid);
		return false;
	}

	private boolean integer7() {
		int number = getInt();
		int dahi = number / 10;
		int vohid = number % 10;
		int sum = dahi + vohid;
		int zarb = dahi * vohid;
		put(sum);
		put(zarb);
	}

	private boolean integer8() {
		int number = getInt();
		int dahi = number / 10;
		int vohid = number % 10;
		number = vohid * 10 + dahi;
		put(number);
		return false;
	}

	private boolean integer9() {
		int number = getInt();
		int sadi = number / 100;
		put(sadi);
		return false;
	}

	private boolean integer10() {
		int number = getInt();
		int vohid = number % 10;
		int dahi = number / 10 % 10;
		put(vohid);
		put(dahi);
		return false;
	}

	private boolean integer11() {
        int number = getInt();
        int sadi = number / 100 % 10;
        int dahi = number / 10 % 10;
        int vohid = number % 10;
        int sum = sadi + dahi + vohid;
        int zarb = sadi * dahi * vohid;
        put(sum);
        put(zarb);
        return false;
    }

	private boolean integer12() {
        int number = getInt();
        int sadi = number / 100 % 10;
        int dahi = number / 10 % 10;
        int vohid = number % 10;
        number = vohid * 100 + dahi * 10 + sadi;
        put(number);
        return false;
    }

	private boolean integer13() {
        int number = getInt();
        int sadi = number / 100 % 10;
        int dahi = number / 10 % 10;
        int vohid = number % 10;
        number = dahi * 100 + vohid * 10 + sadi;
        put(number);
        return false;
    }

	private boolean integer14() {
        int number = getInt();
        int sadi = number / 100 % 10;
        int dahi = number / 10 % 10;
        int vohid = number % 10;
        number = vohid * 100 + sadi * 10 + dahi;
        put(number);
        return false;
    }

	private boolean integer15() {
        int number = getInt();
        int sadi = number / 100 % 10;
        int dahi = number / 10 % 10;
        int vohid = number % 10;
        number = dahi * 100 + sadi * 10 + vohid;
        put(number);
        return false;
    }

	private boolean integer16() {
        int number = getInt();
        int sadi = number / 100 % 10;
        int dahi = number / 10 % 10;
        int vohid = number % 10;
        number = sadi * 100 + vohid * 10 + dahi;
        put(number);
        return false;
    }

	private boolean integer17() {
        int number = getInt();
        int sadi = number / 100 % 10;
        put(sadi);
        return false;
    }

	private boolean integer18() {
        int number = getInt();
        int hazori = number / 1000 % 10;
        put(hazori);
        return false;
    }

	private boolean integer19() {
        int N = getInt();
        int minutes = N / 60;
        put(minutes);
        return false;
    }

	private boolean integer20() {
        int N = getInt();
        int hours = N / 3600;
        put(hours);
        return false;
    }

	private boolean integer21() {
        int N = getInt();
        int seconds = N % 60;
        put(seconds);
        return false;
    }

	private boolean integer22() {
        int N = getInt();
        int seconds = N % 3600;
        put(seconds);
        return false;
    }

	private boolean integer23() {
        int N = getInt();
        int minutes = N % 3600 / 60;
        put(minutes);
        return false;
    }

	private boolean integer24() {
        int K = getInt();
        int weekNo = K % 7;
        put(weekNo);
        return false;
    }

	private boolean integer25() {
        int K = getInt();
        int weekNo = (K + 3) % 7;
        put(weekNo);
        return false;
    }

	private boolean integer26() {
        int K = getInt();
        int weekNo = K % 7 + 1;
        put(weekNo);
        return false;
    }

	private boolean integer27() {
        int K = getInt();
        int weekNo = (K + 4) % 7 + 1;
        put(weekNo);
        return false;
    }

	private boolean integer28() {
        int K = getInt();
        int N = getInt();
        int weekNo = (K + N + 5) % 7 + 1;
        put(weekNo);
        return false;
    }

	private boolean integer29() {
        int A = getInt();
        int B = getInt();
        int C = getInt();
        int kvads = (A / C) * (B / C);
        int freeSpace = A*B - kvads * C * C;
        put(kvads);
        put(freeSpace);
        return false;
    }

	private boolean integer30() {
        int year = getInt();
        int century = (year - 1) / 100 + 1;
        put(century);
        return false;
    }
}